controllers/propagator/replicatedpolicy_controller.go

```diff
@@ -620,13 +622,131 @@ func (r *ReplicatedPolicyReconciler) cleanUpReplicated(ctx context.Context, repl
 
 	deleteErr := r.Delete(ctx, replicatedPolicy)
 
-	version.resourceVersion = "deleted"
+	if deleteErr != nil {
+		if k8serrors.IsNotFound(deleteErr) {
+			version.resourceVersion = "deleted"
+		}
+	} else {
+		version.resourceVersion = "deleted"
 
-	if watcherErr != nil {
-		return watcherErr
+		// Normally the spec-sync controller handles this, however, if it's a self-managed hub policy, the spec-sync
+		// controller does not run. So this is a special case.
+		if replicatedPolicy.Namespace == "local-cluster" && r.ComplianceServerCtx.DB != nil {
+			r.recordDisabledEvents(ctx, replicatedPolicy)
+		}
+	}
+
+	return errors.Join(watcherErr, deleteErr)
+}
+
+// recordDisabledEvents will generate and record disabled compliance events in the compliance history database for each
+// policy template. On retryable errors, the generated compliance event is added to the ComplianceServerCtx queue
```

Comment:
Because MonitorDatabaseConnection sleeps for 20 seconds after each check, which is what monitors the queue. I could have used a channel instead to avoid this, however, a channel has to have a bound limit which could be an issue in a large environment.

Additionally, I prefer do it synchronously in the happy-path to make timing more predictable.